### DISCUSSION TITLE
Support for Device Groups

### DISCUSSION THREAD

**Discussion by mfcodeworks** [kreait/firebase-php#335] [Ideas]
Does this package have support for Device Groups?

https://firebase.google.com/docs/cloud-messaging/android/device-group#managing_device_groups

I can see there's support for multi-cast messages and topics but is there device group support available for creating/updating groups?

Usually I prefer topics for specific topics e.g. order-441 but I'll use Device Groups for a user. If a user logs in on Android, iOS, and web then I add each token to a user device group and I only send to the group token to notify each of their apps

That would be the second part of the request, if targeting a token has support for group tokens? Even if the group token was created with a separate package would I be able to send to it using this package?

**Comment by jeromegamez** [comment]
Not yet, but I‘ll look into it. Consider this an accepted feature request ^^

**Comment by jeromegamez** [comment]
Quick heads-up, I started working on it, but this is only supported via the legacy HTTP protocols which have completely different requirements (a Server Key Key, different message formats).

In my first tests, the Legacy REST API is still rejecting my requests although I've used the API Key and Sender ID of my project.

Even if I manage to get it to work, there will still be the issue that you will not be able to actually send messages to those device groups, because they can only be addressed via the Legacy protocol, whereas this SDK relies on the V1 protocol. The protocols are incompatible, and I can't bring myself to fully support both of them in the SDK.

At this point, I'm tending to postpone this feature until (and if) it's available via the newer protocol :/

**Comment by mfcodeworks** [comment]
So currently impossible to send device group messages with the new SDK? :/
I think for now I'll stick to the legacy API for groups and message sending, SDK for subscription management and admin features
